### PR TITLE
Add ARMC6 where appropriate

### PR DESCRIPTION
**Pull request by theotherjimmy**
# DO NOT MERGE BEFORE https://github.com/ARMmbed/mbed-os/pull/4949 IS MERGED

### CODE REVIEW

File: docs/tutorials/optimize/memory/memap.md [hunk 1]
```diff
@@ -72,7 +72,7 @@ There are other entries that require a bit of clarification:
 
 #### Current support
 
-We have tested *memap* on Windows 7, Linux and Mac OS X. The `GCC_ARM` (GNU Arm Embedded Toolchain), Arm (Arm Compiler 5) and IAR toolchains generate memory map files.
+We have tested *memap* on Windows 7, Linux and Mac OS X. The `GCC_ARM` (GNU Arm Embedded Toolchain), Arm Compiler 5, Arm Compiler 6 and  IAR toolchains generate memory map files.
```
Comment (AnotherButler) [inline docs/tutorials/optimize/memory/memap.md]: Please delete the extra space after "and".

**Comment by AnotherButler** [comment]
@theotherjimmy Please resolve conflicts if this is ready for me to merge.

**Comment by theotherjimmy** [comment]
Rebase resolved it trivially, apparently.

**Comment by theotherjimmy** [comment]
Just a sec, removing out of date info

**Comment by theotherjimmy** [comment]
Done.

**Comment by theotherjimmy** [comment]
Wait. One more to remove.

**Comment by theotherjimmy** [comment]
Sorry for the noise. It's all correct now.

**Pull request merged**